File: src/main/java/me/spartacus04/jext/command/LocationParser.java
package me.spartacus04.jext.command;

import org.bukkit.Location;
import org.bukkit.entity.Player;

public class LocationParser {

  private String stringX;
  private String stringY;
  private String stringZ;
  private Player sender;

  public LocationParser(String stringX, String stringY, String stringZ, Player sender) throws IllegalStateException {
    this.stringX = stringX;
    this.stringY = stringY;
    this.stringZ = stringZ;
    this.sender = sender;
  }

  public Location parse() {
    final int intX;
    if (stringX.equals("~")) {
      intX = sender.getLocation().getBlockX();
    } else {
      intX = (int)Float.parseFloat(stringX);
    }

    final int intY;
    if (stringY.equals("~")) {
      intY = sender.getLocation().getBlockY();
    } else {
      intY = (int)Float.parseFloat(stringY);
    }

    final int intZ;
    if (stringZ.equals("~")) {
      intZ = sender.getLocation().getBlockZ();
    } else {
      intZ = (int)Float.parseFloat(stringZ);
    }

    return new Location(sender.getWorld(), intX, intY, intZ);
  }
}

File: src/main/java/me/spartacus04/jext/command/LocationParser.kt
package me.spartacus04.jext.command

import org.bukkit.Location
import org.bukkit.entity.Player

class LocationParser(
    private val stringX: String,
    private val stringY: String,
    private val stringZ: String,
    private val sender: Player
) {
    fun parse(): Location {
        val intX: Int = if (stringX == "~") {
            sender.location.blockX
        } else {
            stringX.toFloat().toInt()
        }

        val intY: Int = if (stringY == "~") {
            sender.location.blockY
        } else {
            stringY.toFloat().toInt()
        }

        val intZ: Int = if (stringZ == "~") {
            sender.location.blockZ
        } else {
            stringZ.toFloat().toInt()
        }

        return Location(sender.world, intX.toDouble(), intY.toDouble(), intZ.toDouble())
    }
}